server/src/agents/executor/toolkit.ts

```diff
@@ -217,33 +219,100 @@ export const getExecutorToolkit = (account: Account) => {
           return `Transaction not found for task: "${task}" [id: ${taskId}].`;
         }
 
-        const walletClient = createWalletClient({
-          account,
-          chain: getChain(parseInt(env.CHAIN_ID)),
-          transport: http(),
+        const chain = getChain(parseInt(env.CHAIN_ID));
+        if (!chain) return `Chain not supported`;
+
+        // const walletClient = createWalletClient({
+        //   account,
+        //   chain: getChain(parseInt(env.CHAIN_ID)),
+        //   transport: http(),
+        // });
+        // const publicClient = createPublicClient({
+        //   chain: getChain(parseInt(env.CHAIN_ID)),
+        //   transport: http(),
+        // });
+
+        const AGENT_ADDRESS = "0xE5A730337eaF120A7627AB7A3F083a7b4b865EB0";
+        const AGENT_PRIVATE_KEY = env.SAFE_AGENT_PRIVATE_KEY;
+        const HUMAN_SIGNER_1_ADDRESS =
+          "0x4e763f42227DF08696389d4fcA2Df0b5Fe33f246";
+        const HUMAN_SIGNER_2_ADDRESS =
+          "0x58dEe4Cd5d60ed92eC6e5d3b14788aF0819A3698";
+
+        const RPC_URL = chain.rpcUrls.default.http[0];
+
+        const safeClient = await Safe.init({
+          provider: RPC_URL,
+          signer: AGENT_PRIVATE_KEY,
+          predictedSafe: {
+            safeAccountConfig: {
+              owners: [
+                AGENT_ADDRESS,
+                HUMAN_SIGNER_1_ADDRESS,
+                HUMAN_SIGNER_2_ADDRESS,
+              ],
+              threshold: 2,
+            },
+          },
         });
-        const publicClient = createPublicClient({
-          chain: getChain(parseInt(env.CHAIN_ID)),
-          transport: http(),
+
+        console.log("Safe client created>>>>", safeClient);
+
+        const apiKit = new SafeApiKit({
```

Comment:
this chainid should be coming from the prompts and not restricted to an env variable.